test/parallel/test-vm-dynamic-import.js

```diff
@@ -0,0 +1,105 @@
+'use strict';
+
+// Flags: --experimental-vm-modules --experimental-modules
+
+const common = require('../common');
+const assert = require('assert');
+common.crashOnUnhandledRejection();
+
+const vm = require('vm');
+
+async function testModule() {
+  {
+    const module = new vm.Module('import("X")');
+    await module.link(common.mustCall((specifier) => {
+      assert.strictEqual(specifier, 'X');
+      return new vm.Module('export const x = 5');
+    }));
+    module.instantiate();
+    const { result } = await module.evaluate();
+    assert.strictEqual((await result).x, 5);
+  }
+
+  {
+    const module = new vm.Module('import("X")');
+    await module.link(common.mustCall(async (specifier) => {
+      const m = new vm.Module('export const x = 5');
+      await m.link(() => undefined);
+      m.instantiate();
+      await m.evaluate();
+      return m.namespace;
+    }));
+    module.instantiate();
+    const { result } = await module.evaluate();
+    assert.strictEqual((await result).x, 5);
+  }
+}
+
+async function testScript() {
+  let moduleX;
+  const script = new vm.Script('import("vm://X")', {
+    resolveDynamicImport(specifier, scriptOrModule) {
+      if (specifier === 'vm://X') {
+        assert.strictEqual(scriptOrModule, script);
+        return moduleX = new vm.Module('export default import("vm://Y")', { url: 'vm://X' });
+      }
+      if (specifier === 'vm://Y') {
+        assert.strictEqual(scriptOrModule, moduleX);
+        return new vm.Module('export default 5', { url: 'vm://Y' });
+      }
```

Comment:
@devsnek neither of the above sources have static imports, so I don't think this covers the unlinked test case I described?

Reply:
it will also inherit the resolveDynamicImport function for that which makes me think we need to change the name again

Reply:
Surely `resolveDynamicImport` should never be inherited from Script to Module, only the linker should be inherited from Module to Module?

Reply:
I think the hang-up we are having is differentiating scripts and modules too much and differentiating dynamic linking and static linking too much. if we just have "linker" and document that it handles all the things I think it should be fine